keylime/registrar_client.py

```diff
@@ -64,7 +90,20 @@ def getData(registrar_ip, registrar_port, agent_id, tls_context):
             logger.critical("Error: did not receive port from Registrar Server.")
             return None
 
-        return response_body["results"]
+        r = response_body["results"]
+        res: RegistrarData = {
+            "aik_tpm": r["aik_tpm"],
```

Comment:
Is this working with pyright?

I was expecting more like: `"aik_tpm": bytes(r["aik_tpm"], encoding="utf-8"), "port": int(r["port"]), ...`

Reply:
Yes, pyright passes.

Reply:
That is wild then!

A JSON should be considered mostly a `dict[str, Any]`, hidden in some C wraps. I guess that there is some thing that I do not understand in the subtyping / variance / contra-variance algorithm used in pyright (or less probably that in some time it will complain here).

Anyway :+1:

Reply:
> That is wild then!

Indeed! I was surprised here as well that it take entries from a map where it doesn't know what the individual values' data types are and doesn't at least require a cast.